### PR TITLE
Fix workflow integration tests for pr-agent and local actions

### PR DESCRIPTION
**Pull request by mohavro**
## Summary
- align pr-agent workflow tests with actual workflow name and job structure
- allow local composite actions without version tags in workflow action version checks
- fix environment variable naming test indentation and cleanup duplicated logic

## Testing
- python -m pytest tests/integration/test_github_workflows.py -k "pr_agent_name or pr_agent_trigger_runs_on_ubuntu or workflow_actions_have_versions or workflow_env_vars_naming_convention" -q

[Codex Task](https://chatgpt.com/codex/tasks/task_e_6921d789639c832482b6e40e4faa6d71)

## Summary by Sourcery

Update GitHub workflow integration tests for action version checks and PR Agent workflow validation.

Bug Fixes:
- Fix workflow action version tests to correctly allow additional forms of local composite actions without version tags.
- Align PR Agent workflow tests with the actual workflow name and ensure the trigger job is correctly validated and deduplicated.
- Correct the environment variable naming convention test by fixing indentation and removing duplicated logic.

## Summary by cubic
Fixes GitHub workflow integration tests to match the PR Agent workflow and tighten action pinning. Adds validation tests for pr-agent config and workflow simplifications, plus unit tests for auth, database, app, asset graph, and the frontend API client.

- **Bug Fixes**
  - Align pr-agent tests with the workflow name containing "PR Agent" and the "pr-agent-trigger" job.
  - Verify the trigger job runs on standard Ubuntu runners.
  - Require pinned versions for all actions and block floating branches; remove the local-action exemption.
  - Fix duplicated and mis-indented env var naming validation, validating top-level env once.
  - Fail the workflow consistency test on YAML load errors; skip only when a workflow file is missing.
  - Assert greetings.yml triggers when "on" is a mapping or uses the boolean shorthand (on: true) so required events can’t be silently skipped.

- **New Tests**
  - Add tests for pr-agent-config.yml structure and removal of obsolete chunking settings.
  - Add workflow tests to confirm simplifications (no duplicate Setup Python, no chunking steps, no orphaned script references, no label/config existence checks, no credential checks).
  - Add comprehensive unit tests for auth, database, app, asset graph, and the frontend API client.

<sup>Written for commit ae0767ed52703f1e648c361ba89987f653761e94. Summary will update on new commits. <a href="https://cubic.dev/pr/DashFin-FarDb/financial-asset-relationship-db/pull/260">Review in cubic</a></sup>

### **Auto-created Ticket**

[#636](https://github.com/DashFin-FarDb/financial-asset-relationship-db/issues/636)

### **Description**
- Added comprehensive unit tests for the `api/auth.py` authentication module, covering various aspects including password hashing, user repository operations, and JWT token handling.
- Introduced extensive tests for the Gradio application in `app.py`, focusing on initialization, metrics, and UI interactions.
- Developed thorough tests for the `AssetRelationshipGraph` class, ensuring consistent behavior and edge case handling.
- Implemented tests for refactored authentication methods in `api/auth.py`, validating parameter changes and thread safety.
- Created tests for frontend API client refactoring, focusing on type safety and error handling.
- Documented the additions and generated reports summarizing the comprehensive test generation process.

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_auth.py</strong><dd><code>Comprehensive Unit Tests for Authentication Module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/unit/test_auth.py
<li>Added comprehensive unit tests for the <code>api/auth.py</code> authentication <br>module.<br> <li> Included tests for password hashing, user repository operations, JWT <br>token creation, and user authentication flows.<br> <li> Covered edge cases and security scenarios.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-b64a2624bf10a6799dd1c939b236250ff669d6cb08d17daadb82c59ce32406fd">+879/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_app.py</strong><dd><code>Comprehensive Unit Tests for Gradio Application</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/unit/test_app.py
<li>Added comprehensive unit tests for the Gradio application in <code>app.py</code>.<br> <li> Tested application initialization, metrics calculation, asset <br>information retrieval, and UI interactions.<br> <li> Included edge cases and error handling scenarios.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-fcbbe89d576bb1122eba8970740c18a3f3eb876b634ed6b62789d7561cae8372">+773/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_asset_graph.py</strong><dd><code>Comprehensive Unit Tests for Asset Relationship Graph</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/unit/test_asset_graph.py
<li>Added extensive unit tests for the <code>AssetRelationshipGraph</code> class.<br> <li> Tested various scenarios including single relationships, circular <br>layouts, and edge cases.<br> <li> Ensured consistent behavior across multiple calls and edge cases.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-f69944b4d3462deae12ddfb8d345455bf356c3be87c16c4b4f1896082d985360">+504/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_auth_refactoring.py</strong><dd><code>Unit Tests for Auth Refactoring Changes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/unit/test_auth_refactoring.py
<li>Added tests for refactored authentication module focusing on <br>UserRepository instance methods.<br> <li> Validated parameter name changes and edge cases.<br> <li> Ensured thread safety of repository methods.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-a646ae904dc71aedd87d98075cb57db72c1da016798a07933b4dda9479bc4930">+619/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>api-refactoring.test.ts</strong><dd><code>Frontend API Client Refactoring Tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/__tests__/lib/api-refactoring.test.ts
<li>Added tests for frontend API client refactoring.<br> <li> Focused on quote style consistency, type definitions, parameter <br>formatting, and error handling.<br> <li> Validated response type and edge cases.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-7bf7a99d9cd91b9772308f7981e1d2506708d1a51c81e8b87b76027153b5eeec">+392/-0</a>&nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>TEST_ADDITIONS_SUMMARY.md</strong><dd><code>Summary of New Test Additions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

TEST_ADDITIONS_SUMMARY.md
<li>Summarized comprehensive unit tests added for workflow <br>simplifications.<br> <li> Detailed new test classes and their purposes.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-5e19ca73c4c53e871d08d52aca53b93208221b9b5c2c5aa79b67e2f475498dc0">+126/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>COMPREHENSIVE_TEST_GENERATION_REPORT.md</strong><dd><code>Comprehensive Test Generation Report</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

COMPREHENSIVE_TEST_GENERATION_REPORT.md
<li>Report on the comprehensive test generation process and results.<br> <li> Included statistics on new test files, lines of code, and test <br>methods.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260/files#diff-e6b6f07135d2e53405ab96019f32c26eb44421b3023890055f9cd87120185f79">+30/-0</a>&nbsp; &nbsp; </td>
</tr>
</table></td></tr></tr></tbody></table>

> 💡 **Penify usage**:
>Comment `/help` on the PR to get a list of all available Penify tools and their descriptions

___

## **CodeAnt-AI Description**
Add comprehensive unit and frontend tests plus test validation and quick-run docs

### What Changed
- Added large suites of unit tests covering authentication, app UI logic, and asset-relationship graph behavior so auth, app, and visualization logic are exercised in many edge cases
- Added frontend API client tests that validate request URLs, parameter handling, error propagation, and concurrent request behavior
- Added test-validation script and a quick-reference README to verify generated tests and run common test groups locally
- Updated integration workflow tests to align PR Agent workflow naming and allow local composite actions without pinned versions

### Impact
`✅ More test coverage for auth, app, and graph`
`✅ Fewer regressions in frontend API client behavior`
`✅ Easier local validation and execution of generated tests`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: FINAL_TEST_GENERATION_SUCCESS_SUMMARY.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+## 🚀 Running the Tests
+
+### Quick Start
+```bash
+pytest tests/integration/test_pr_agent_config.py \
+       tests/integration/test_workflow_simplifications.py -v
+```
+
+**Expected Output**:
```
[the remark this answers is not in the document]
Comment (mohavro) [reply FINAL_TEST_GENERATION_SUCCESS_SUMMARY.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/260#discussion_r2703051567)

File: TEST_GENERATION_COMPLETE_CURRENT_BRANCH.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+- `.github/labeler.yml` - Configuration file removed
+- `.github/scripts/README.md` - Documentation removed
+- `.github/scripts/context_chunker.py` - Python script removed
+
+### 4. Test Files (Already Added)
+- Multiple test files in `frontend/__tests__/` and `tests/integration/`
+
+## New Tests Generated
+
+### 1. tests/integration/test_pr_agent_config.py (NEW - 487 lines)
```
[the remark this answers is not in the document]
Comment (mohavro) [reply TEST_GENERATION_COMPLETE_CURRENT_BRANCH.md]: @copilot fix this

File: TEST_GENERATION_COMPLETE_CURRENT_BRANCH.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+- Consistent action versions
+- Reasonable workflow sizes
+- ✅ No orphaned script references (context_chunker.py)
+
+## Test Statistics
+
+| Metric | Value |
+|--------|-------|
+| **New Test Files Created** | 2 |
+| **Total Lines of Test Code** | 1,128 lines |
```
[the remark this answers is not in the document]
Comment (mohavro) [reply TEST_GENERATION_COMPLETE_CURRENT_BRANCH.md]: @copilot fix this
